src/ILCompiler.Compiler/src/Compiler/CompilerTypeSystemContext.cs

```diff
@@ -238,7 +238,7 @@ private EcmaModule AddModule(string filePath, string expectedSimpleName)
                     ModuleData actualModuleData = _simpleNameHashtable.AddOrGetExisting(moduleData);
                     if (actualModuleData != moduleData)
                     {
-                        if (actualModuleData.FilePath != filePath)
+                        if (actualModuleData.FilePath != filePath && actualModuleData.SimpleName != "mscorlib")
```

Comment:
The fix will need to happen somewhere in `dotnet` or `dotnet-compile-native`. I think a hack is okay for now if you put it in the right spot.